File: Arithmetic Problems/Birthday_Paradox/Birthday_Paradox.cpp
// Program to approximate number of people in Birthday Paradox problem

#include <cmath>
#include <iostream>
using namespace std;

// Returns approximate number of people for a given probability
int find(double p)
{
    return ceil(sqrt(2*365*log(1/(1-p))));
}

int main()
{
    printf("%d",find(0.70));
}

/* Output
3o
*/

File: Birthday_Paradox/Birthday_Paradox.java
//Java program to approximate number of people in Birthday Paradox problem
class Birthday_Paradox 
{

    static double find(double p) // Returns approximate number of people for a given probability
    {
        return Math.ceil(Math.sqrt(2*365*Math.log(1/(1-p))));
    }

    public static void main(String[] args)
    {
        System.out.println(find(0.70));
    }
}

/* Output
30.0
*/
